scripts/linters/python_linter.py

```diff
@@ -115,20 +117,140 @@ def _check_import_order(self):
+
+        one_off_jobs_list = []
+        validation_jobs_list = []
+        for filepath in self.all_filepaths:
+            if filepath.endswith('prod_validation_jobs_one_off.py'):
+                validation_jobs_list.extend(_get_jobs_class_names_in_filepath(
+                    filepath, 'ProdValidationAuditOneOffJob'))
+            elif filepath.endswith('_jobs_one_off.py'):
+                one_off_jobs_list.extend(_get_jobs_class_names_in_filepath(
+                    filepath, 'BaseMapReduceOneOffJobManager'))
+
+        # Removing jobs which are used in cron.
+        one_off_jobs_list = [
+            job for job in one_off_jobs_list if job not in jobs_in_cron]
+        one_off_jobs_set = set(one_off_jobs_list)
+        if len(one_off_jobs_list) != len(one_off_jobs_set):
+            failed = True
+            duplicate_one_off_job_names = (
+                linter_utils.get_duplicates_from_list_of_strings(
+                    one_off_jobs_list))
+            summary_message = 'Found one-off jobs with duplicate names: %s' % (
+                ', '.join(duplicate_one_off_job_names))
+            python_utils.PRINT(summary_message)
+            summary_messages.append(summary_message)
+
+        if validation_jobs_list:
+            # Removes the base validation job class the list.
+            validation_jobs_list.remove('ProdValidationAuditOneOffJob')
+        validation_jobs_set = set(validation_jobs_list)
+        if len(validation_jobs_list) != len(validation_jobs_set):
+            failed = True
+            duplicate_validation_job_names = (
+                linter_utils.get_duplicates_from_list_of_strings(
+                    validation_jobs_list))
+            summary_message = (
+                'Found validation jobs with duplicate names: %s' % (
+                    ', '.join(duplicate_validation_job_names)))
+            python_utils.PRINT(summary_message)
+            summary_messages.append(summary_message)
+
+        non_registered_one_off_jobs = (
+            one_off_jobs_set - expected_one_off_jobs_set)
+        if non_registered_one_off_jobs:
+            failed = True
+            summary_message = (
+                'Found one-off jobs not listed in jobs_registry file: %s' % (
+                    ',\n'.join(sorted(non_registered_one_off_jobs))))
+            python_utils.PRINT(summary_message)
+            summary_messages.append(summary_message)
+
+        non_registered_validation_jobs = (
+            validation_jobs_set - expected_validation_jobs_set)
+        if non_registered_validation_jobs:
+            failed = True
+            summary_message = (
+                'Found validation jobs not listed in jobs_registry file: %s' % (
+                    ',\n'.join(sorted(non_registered_validation_jobs))))
+            python_utils.PRINT(summary_message)
+            summary_messages.append(summary_message)
+
+        if failed:
+            summary_message = (
```

Comment:
Always try to reduce branching. In this case only the "failed"/"passed" string differs so get rid of the if/else and just substitute that part with `'failed' if failed else 'passed'`.

Reply:
I agree! (I was following the existing patterns) Fixed!